src/systems/a320_systems/src/hydraulic/mod.rs

```diff
@@ -2825,6 +2956,299 @@ impl SimulationElement for A320HydraulicOverheadPanel {
+    const ALL_POWER_BUSES: [ElectricalBusType; 4] = [
+        ElectricalBusType::DirectCurrentEssential,
+        ElectricalBusType::DirectCurrent(2),
+        ElectricalBusType::DirectCurrentHot(1),
+        ElectricalBusType::DirectCurrentHot(2),
+    ];
+
+    fn new(context: &mut InitContext) -> Self {
+        Self {
+            requested_position_left_id: context
+                .get_identifier("HYD_AILERON_LEFT_DEMAND".to_owned()),
+            requested_position_right_id: context
+                .get_identifier("HYD_AILERON_RIGHT_DEMAND".to_owned()),
+
+            left_position_requested: Ratio::default(),
+            right_position_requested: Ratio::default(),
+
+            // Controllers are in outboard->inboard order, so for aileron [Blue circuit, Green circuit]
+            left_controllers: [AileronController::new(), AileronController::new()],
+            right_controllers: [AileronController::new(), AileronController::new()],
+
+            is_powered: false,
+        }
+    }
+
+    fn update_aileron_requested_position(&mut self) {
+        for controller in &mut self.left_controllers {
+            controller.set_requested_position(self.left_position_requested);
+        }
+
+        for controller in &mut self.right_controllers {
+            controller.set_requested_position(self.right_position_requested);
+        }
+    }
+
+    fn set_aileron_no_position_control(&mut self) {
+        for controller in &mut self.left_controllers {
+            controller.set_mode(LinearActuatorMode::ClosedCircuitDamping);
+        }
+
+        for controller in &mut self.right_controllers {
+            controller.set_mode(LinearActuatorMode::ClosedCircuitDamping);
+        }
+    }
+
+    fn set_left_aileron_position_control(
+        &mut self,
+        hydraulic_configuration: AileronHydConfiguration,
+    ) {
+        match hydraulic_configuration {
+            AileronHydConfiguration::GB | AileronHydConfiguration::B => {
+                self.left_controllers[AileronActuatorCircuit::Blue as usize]
+                    .set_mode(LinearActuatorMode::PositionControl);
+                self.left_controllers[AileronActuatorCircuit::Green as usize]
+                    .set_mode(LinearActuatorMode::ActiveDamping);
+            }
+
+            AileronHydConfiguration::G => {
+                self.left_controllers[AileronActuatorCircuit::Blue as usize]
+                    .set_mode(LinearActuatorMode::ActiveDamping);
+                self.left_controllers[AileronActuatorCircuit::Green as usize]
+                    .set_mode(LinearActuatorMode::PositionControl);
+            }
+            _ => {
+                self.left_controllers[AileronActuatorCircuit::Blue as usize]
+                    .set_mode(LinearActuatorMode::ClosedCircuitDamping);
+                self.left_controllers[AileronActuatorCircuit::Green as usize]
+                    .set_mode(LinearActuatorMode::ClosedCircuitDamping);
+            }
```

Comment:
I believe we try to handle the different cases for match statements explicitly, rather than using this fallback `_`. Even if this is actually exhaustive for the enum member we have now.